torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1060,6 +1057,41 @@ def test_all_reduce_sum_cuda(self):
                 rank_to_GPU,
             )
 
+        @unittest.skipIf(BACKEND == "nccl", "Nccl does not support CPU tensors")
+        def test_all_reduce_sum_complex(self):
```

Comment:
I only bothered adding complex tests for a single reduction op (`sum`) since it's unrelated to the actual op logic

Reply:
update to that since the above comment isn't really true- added explicit tests that a complex-unsupported reduceOp like `Max` should error out properly :)